archinstall/__init__.py

```diff
@@ -280,8 +282,20 @@ def plugin(f, *args, **kwargs):
 
 def _check_new_version():
 	info("Checking version...")
-	Pacman.run("-Sy")
-	upgrade = Pacman.run("-Qu archinstall").decode()
+
+	try:
+		Pacman.run("-Sy")
+	except Exception as e:
+		debug(f'Failed to perform version check: {e}')
+		info(f'Arch Linux mirrors are not reachable. Please check your internet connection')
+		exit(1)
+
+	upgrade = None
+
+	try:
+		upgrade = Pacman.run("-Qu archinstall").decode()
+	except Exception as e:
+		debug(f'Failed determine pacman version: {e}')
```

Comment:
Could we do [pip check](https://stackoverflow.com/a/62571316) as backup?

Reply:
I think the stated intent is met without that.

> major relevance for the version check is for folks running it in a old ISO

In my development environment I have archinstall installed in [development mode](https://setuptools.pypa.io/en/latest/userguide/development_mode.html) (dependencies installed via pacman and substitue in `python -m venv --system-site-packages .venv` when creating venv). This is a pip install and I would rather not have to remember the new `--skip-version-check` argument to prevent sending a HTTP request to PyPI everytime I launched the installer to do testing.

Reply:
Fair enough, I think we could get around it by checking if we're [in venv](https://stackoverflow.com/a/1883251) or not.

But I think the main goal is achieved here by eliminating users running old versions on the official ISO's :)